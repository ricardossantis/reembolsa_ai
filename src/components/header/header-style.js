import styled from "styled-components";
import { Link } from "react-router-dom";
import { Menu } from "antd";
import { MenuOutlined } from "@ant-design/icons";

export const MaxHeader = styled.div`
  display: flex;
  flex-wrap: wrap;
  justify-content: space-evenly;
  align-items: center;
  width: 100%;
  height: 97px;
  color: #ffffff;
  font-size: 24px;
`;

export const MinHeader = styled.div`
  display: flex;
  flex-wrap: wrap;
  justify-content: flex-end;
  align-items: center;
  width: 100%;
  height: 97px;
`;

export const ZLink = styled(Link)`
  color: #ffffff;
  :hover {
    color: #2cd3b5;
  }
`;
export const MLink = styled(Link)`
  color: #ffffff;
`;

export const Hamburguer = styled(MenuOutlined)`
  margin-right: 1.5rem;
  font-size: 3rem;
  :hover {
    color: #2cd3b5;
  }
`;
export const ZMenu = styled(Menu)`
  background-color: #365083;
`;

export const MenuItem = styled(Menu.Item)`
  padding: 1.5rem;
  color: #ffffff;
  font-size: 24px;

  :hover {
    background-color: #2CD3B5;
  }
`;
