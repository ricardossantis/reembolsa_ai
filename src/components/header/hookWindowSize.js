import { useState, useEffect } from "react";

export const useWindowSize = () => {
  const initialState = {
    width: 0,
    height: 0,
  };

  const [windowSize, setWindowSize] = useState(initialState);

  useEffect(() => {
    const handleSize = () => {
      setWindowSize({
        width: window.innerWidth,
        height: window.innerHeight,
      });
    };
    window.addEventListener("resize", handleSize);
    handleSize();
    return () => window.removeEventListener("resize", handleSize);
  }, []);
  return windowSize;
};
