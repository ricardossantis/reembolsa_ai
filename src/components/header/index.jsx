import React from "react";
import styled from "styled-components";

const Header = () => {
  return <div>teste</div>;
};

export default Header;
