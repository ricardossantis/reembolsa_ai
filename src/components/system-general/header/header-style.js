import styled from "styled-components";
import { Link } from "react-router-dom";
import { Menu } from "antd";
import { MenuOutlined } from "@ant-design/icons";

export const MaxHeader = styled.div`
  position: sticky;
  top: 0;
  right: 0;
  left: 0;
  z-index: 1030;
  display: flex;
  flex-wrap: wrap;
  justify-content: space-evenly;
  align-items: center;
  height: 97px;
  color: #ffffff;
  font-size: 24px;
`;

export const MinHeader = styled.div`
  position: sticky;
  top: 0;
  right: 0;
  left: 0;
  z-index: 1030;
  display: flex;
  flex-wrap: wrap;
  justify-content: space-around;
  align-items: center;
  padding: 1rem;
  height: 97px;
`;

export const ZLink = styled(Link)`
  color: #ffffff;
  :hover {
    color: #2cd3b5;
  }
`;
export const LogoLink = styled(Link)`
  @import url("https://fonts.googleapis.com/css2?family=Cinzel&display=swap");
  font-family: "Cinzel", serif;
  text-decoration-line: underline;
  color: #ffffff;
  :hover {
    text-decoration-line: underline;
    color: #2cd3b5;
  }
`;
export const MLink = styled(Link)`
  color: #ffffff;
`;

export const Hamburguer = styled(MenuOutlined)`
  font-size: 3rem;
  :hover {
    color: #2cd3b5;
  }
`;
export const ZMenu = styled(Menu)`
  background-color: #365083;
`;

export const MenuItem = styled(Menu.Item)`
  padding: 1.5rem;
  color: #ffffff;
  font-size: 24px;

  :hover {
    background-color: #2cd3b5;
  }
`;
