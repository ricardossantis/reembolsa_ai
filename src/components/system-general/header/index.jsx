import React from "react";
import { useWindowSize } from "./hookWindowSize.js";
import { Dropdown } from "antd";
import {
  MaxHeader,
  MinHeader,
  ZLink,
  LogoLink,
  MLink,
  Hamburguer,
  ZMenu,
  MenuItem,
} from "./header-style.js";

const Header = (props) => {
  const width = useWindowSize().width;

  const menuHamburguer = (
    <ZMenu>
      <MenuItem key="1">
        <ZLink to={props.link1 || props.home} style={{ color: "#FFFFFF" }}>
          {width <= 1066 ? props.title1 : props.logo}
        </ZLink>
      </MenuItem>
      <MenuItem key="2">
        <MLink to={props.link2} style={{ color: "#FFFFFF" }}>
          {props.title2}
        </MLink>
      </MenuItem>
      <MenuItem key="3">
        <MLink to={props.link3} style={{ color: "#FFFFFF" }}>
          {props.title3}
        </MLink>
      </MenuItem>
      <MenuItem key="4">
        <MLink
          to={props.link4}
          onClick={props.logoutColab}
          style={{ color: "#FFFFFF" }}
        >
          {props.title4 || props.titleColab}
        </MLink>
      </MenuItem>
      {props.link5 && (
        <MenuItem key="5">
          <MLink to={props.link5} style={{ color: "#FFFFFF" }}>
            {props.title5}
          </MLink>
        </MenuItem>
      )}
      {props.exitManager && (
        <MenuItem key="6">
          <MLink to="/" onClick={props.logout} style={{ color: "#FFFFFF" }}>
            {props.exitManager}
          </MLink>
        </MenuItem>
      )}
    </ZMenu>
  );

  return (
    <>
      {(width > 1066 && (
        <MaxHeader style={{ backgroundColor: props.maxColor }}>
          {props.input}
          {props.logo && (
            <LogoLink to={props.home || "/"}>
              <img src={props.logo} alt="logo" />
            </LogoLink>
          )}

          <ZLink to={props.link1 || "/"}>{props.title1}</ZLink>
          <ZLink to={props.link2}>{props.title2}</ZLink>
          <ZLink to={props.link3}>{props.title3}</ZLink>
          <ZLink to={props.link4}>{props.title4}</ZLink>
          {props.link5 && <ZLink to={props.link5}>{props.title5}</ZLink>}
          <ZLink to="/login" onClick={props.logout} />
          <ZLink to={props.exit ? "/" : "/cadastro"} onClick={props.logout}>
            {props.exit || props.homeButton}
          </ZLink>
        </MaxHeader>
      )) || (
        <MinHeader style={{ backgroundColor: props.minColor }}>
          {props.input}
          <Dropdown overlay={menuHamburguer}>
            <Hamburguer style={{ color: props.burguerColor }} />
          </Dropdown>
        </MinHeader>
      )}
    </>
  );
};

export default Header;
