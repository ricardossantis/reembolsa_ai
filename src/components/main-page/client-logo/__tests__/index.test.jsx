import React from "react";
import renderer from "react-test-renderer";
import ClientLogo from "../index";
import { useWindowSize } from "../../../system-general/header/hookWindowSize";

const clientLogo = {
  logo1: "logo1",
  logo2: "logo2",
  logo3: "logo3",
  logo4: "logo4",
  logo5: "logo5",
  logo6: "logo6",
  logo7: "logo7",
  logo8: "logo8",
};

jest.mock("../../../system-general/header/hookWindowSize", () => ({
  useWindowSize: jest.fn(),
}));

it("width < 768", () => {
  useWindowSize.mockImplementation(() => {
    return { width: 767 };
  });
  const tree = renderer.create(<ClientLogo {...clientLogo} />).toJSON();
  expect(tree).toMatchSnapshot();
});

it("width > 768", () => {
  useWindowSize.mockImplementation(() => {
    return { width: 769 };
  });
  const tree = renderer.create(<ClientLogo {...ClientLogo} />).toJSON();
  expect(tree).toMatchSnapshot();
});
