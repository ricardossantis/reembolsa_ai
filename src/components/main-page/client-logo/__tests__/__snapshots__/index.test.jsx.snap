// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`width < 768 1`] = `
<div
  className="sc-bdnylx hcsGkL"
>
  <div
    className="sc-dlnjPT fFNmza"
  >
    <div
      className="sc-hKFyIo sc-jSFkmK bVKKHk bCCQMK"
    >
      <div
        className="sc-eCApGN sc-gKAblj iFoeTt buyUwi"
      >
        logo1
      </div>
    </div>
    <div
      className="sc-hKFyIo sc-jSFkmK bVKKHk bCCQMK"
    >
      logo2
    </div>
  </div>
  <div
    className="sc-dlnjPT fFNmza"
  >
    <div
      className="sc-hKFyIo sc-jSFkmK bVKKHk bCCQMK"
    >
      <div
        className="sc-eCApGN sc-gKAblj iFoeTt buyUwi"
      >
        logo1
      </div>
    </div>
    <div
      className="sc-hKFyIo sc-jSFkmK bVKKHk bCCQMK"
    >
      logo2
    </div>
  </div>
</div>
`;

exports[`width > 768 1`] = `
<div
  className="sc-bdnylx hcsGkL"
>
  <div
    className="sc-gtssRu hLfcnp"
  >
    <div
      className="sc-hKFyIo bVKKHk"
    >
      <div
        className="sc-eCApGN iFoeTt"
      />
    </div>
    <div
      className="sc-hKFyIo bVKKHk"
    />
    <div
      className="sc-hKFyIo bVKKHk"
    />
    <div
      className="sc-hKFyIo bVKKHk"
    />
  </div>
</div>
`;
